### PR TITLE
Use '-Werror' instead of '-Werror=...' for Jerry libc

### PR DESCRIPTION
**Pull request by LaszloLango**
See also: #718, #719

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -273,7 +273,11 @@ project (Jerry C ASM)
   add_jerry_compile_warnings(all extra format-nonliteral init-self conversion sign-conversion format-security missing-declarations pedantic)
   add_jerry_compile_flags(-Wno-stack-protector -Wno-attributes)
   if(CMAKE_COMPILER_IS_GNUCC)
-    add_jerry_compile_warnings(logical-op)
+    if(${USE_JERRY_LIBC})
+      add_jerry_compile_flags(-Werror -Wlogical-op)
+    else()
+      add_jerry_compile_warnings(logical-op)
+    endif()
```
Comment (akiss77) [inline CMakeLists.txt]: Why the if-else construct? Cannot we just leave `add_jerry_compile_warnings(logical-op)` in place and add
```
if(${USE_JERRY_LIBC})
 add_jerry_compile_flags(-Werror)
else()
```

**Comment by LaszloLango** [comment]
@akiss77, updated

**Comment by akiss77** [comment]
LGTM (FWIW)

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**